### PR TITLE
ECOPROJECT-630: Updated documentation for the standalone NMO

### PR DESCRIPTION
**Pull request by abhatt-rh**
https://issues.redhat.com/browse/TELCODOCS-512 Documents some late breaking changes for the standalone Node Maintenance Operator. Adds the following two modules and removes two modules that are now obsolete.

Applies to OpenShift version : 4.10 +

Preview: 
[Setting a node to maintenance mode by using the web console](https://deploy-preview-42793--osdocs.netlify.app/openshift-enterprise/latest/nodes/nodes/eco-node-maintenance-operator.html#eco-setting-node-maintenance-web-console_node-maintenance-operator)
[Resuming a node from maintenance mode by using the web console](https://deploy-preview-42793--osdocs.netlify.app/openshift-enterprise/latest/nodes/nodes/eco-node-maintenance-operator.html#eco-resuming-node-maintenance-web-console_node-maintenance-operator )

### CODE REVIEW

**Comment by abhatt-rh** [comment]
Hi @razo7 @anna-savina
Kindly review the preview links in this PR that document the recent changes for the standalone NMO. 
cc: @beekhof

File: modules/eco-create-nodemaintenance-cr-web-console.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+.Procedure
+
+. From the *Administrator* perspective in the web console, navigate to *Operators* → *Installed Operators*.
+
+. Select the Node Maintenance Operator from the list of operators.
+
+. In the *Node Maintenance* tab, click *Create NodeMaintenance*.
+
+. In the *Create NodeMaintenance* page, select to configure the `NodeMaintenance` CR via *Form view* or *YAML view*.
```
Comment (bobfuru) [inline modules/eco-create-nodemaintenance-cr-web-console.adoc:24]: "select to configure" -- this phrasing is unclear to me. Select _what_ to configure...? Not sure if this is exactly the meaning you intend, but perhaps something like: 
> ...select `NodeMaintenance` via the **Form view** or **YAML view** to configure the `NodeMaintenance` CR.

File: modules/eco-create-nodemaintenance-cr-web-console.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+. To create the `NodeMaintenance` CR from via *Form view* or *YAML view*, click *Create*.
```
Comment (bobfuru) [inline modules/eco-create-nodemaintenance-cr-web-console.adoc:26]: Can this be combined with the previous step? I'm assuming the process is that the user selects the CR via the Form/YAML view and clicks Create? If so, maybe something like this?
> . To configure the `NodeMaintenance` CR from the **Create NodeMaintenance** page, select `NodeMaintenance` via the **Form view** or **YAML view** and click **Create**.

File: modules/eco-delete-nodemaintenance-cr-web-console.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-resuming-node-maintenance-cr-web-console_{context}"]
+= Resuming a node from maintenance mode in the web console
```
Comment (bobfuru) [inline modules/eco-delete-nodemaintenance-cr-web-console.adoc:7]: I'm getting confused trying to understand why there are multiple modules in the PR that are named either "Resuming a node from maintenance mode in the web console" or "Resuming a node from maintenance mode by using the web console". Is that intentional? In the preview output, I am only seeing the latter version. It looks to me like this is another module file that isn't being included anywhere in the assembly. Please double-check and if necessary, revise the PR to remove the module files that shouldn't be part of the PR.

File: modules/eco-delete-nodemaintenance-cr-web-console.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+.Prerequisites
+
+* Log in as a user with `cluster-admin` privileges.
+* Install the Node Maintenance Operator from the *OperatorHub*.
+
+.Procedure
+
+. From the *Administrator* perspective in the web console, navigate to *Operators* → *Installed Operators*.
+
+. Select the Node Maintenance Operator from the list of operators.
```
Comment (bobfuru) [inline modules/eco-delete-nodemaintenance-cr-web-console.adoc:20]: s/operators/Operators

File: modules/eco-resuming-node-maintenance-cr-web-console.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-resuming-node-maintenance-web-console_{context}"]
+= Resuming a node from maintenance mode by using the web console
```
Comment (bobfuru) [inline modules/eco-resuming-node-maintenance-cr-web-console.adoc:7]: This is the only module with web console instructions that I see appearing in the rendered preview output. (See my previous comment about a similar module that contains this info.)

File: modules/eco-resuming-node-maintenance-cr-web-console.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+.Prerequisites
+
+* Log in as a user with `cluster-admin` privileges.
+* Install the Node Maintenance Operator from the *OperatorHub*.
+
+.Procedure
+
+. From the *Administrator* perspective in the web console, navigate to *Operators* → *Installed Operators*.
+
+. Select the Node Maintenance Operator from the list of operators.
```
Comment (bobfuru) [inline modules/eco-resuming-node-maintenance-cr-web-console.adoc:20]: s/operators/Operators

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 7]
```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+//nodes/nodes/eco-node-maintenance-operator.adoc
+
+:_content-type: PROCEDURE
+[id="eco-setting-node-maintenance-cr-cli_{context}"]
+= Setting a node to maintenance mode by using the CLI
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:7]: It seems like a similar module file (`modules/eco-setting-node-to-maintenance-mode-with-cr.adoc`) is also in this PR.

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+You can put a node into maintenance mode with a `NodeMaintenance` custom resource (CR). When you apply a `NodeMaintenance` CR, all allowed pods are evicted and the node is rendered unschedulable. Evicted pods are queued to be moved to another node in the cluster.
+
+.Prerequisites
+
+* Install the {product-title} CLI `oc`.
+* Log in to the cluster as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create the following node maintenance CR, and save the file as `nodemaintenance-cr.yaml`:
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:18]: The name of the literal CR here should be in backticks:
```suggestion
. Create the following `NodeMaintenance` CR, and save the file as `nodemaintenance-cr.yaml`:
```

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+kind: NodeMaintenance
+metadata:
+  name: maintenance-example  <1>
+spec:
+  nodeName: node-1.example.com <2>
+  reason: "NIC replacement" <3>
+----
+<1> The name of the node maintenance CR
+<2> The name of the node to be put into maintenance mode
+<3> A plain text description of the reason for maintenance
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:32]: These are all full sentences and thus should end with periods.

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
++
+.  Apply the node maintenance CR by running the following command:
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:34]:
```suggestion
.  Apply the `NodeMaintenance` CR by running the following command:
```

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
++
+[source,terminal]
+----
+$ oc apply -f nodemaintenance-cr.yaml
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:38]: Should this filename be the same as the filename you used in the step 1 example (`maintenance-example`)?

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+----
+
+. Check the progress of the maintenance task by running the following command, replacing `<node-name>` with the name of your node:
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:41]: IIUC, you could show the node name you used in the step 1 example (`node-1.example.com`) for consistency:
```suggestion
. Check the progress of the maintenance task by running the following command, replacing `<node-name>` with the name of your node; for example, `node-1.example.com`:
```

File: modules/eco-setting-node-maintenance-cr-cli.adoc [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
++
+[source,terminal]
+----
+$ oc describe node <node-name>
```
Comment (bobfuru) [inline modules/eco-setting-node-maintenance-cr-cli.adoc:45]: If you used the example name in my previous suggestion, you'd want it here, too. Your call. :)

File: nodes/nodes/eco-node-maintenance-operator.adoc [hunk 14]
```diff
@@ -29,11 +29,11 @@ The Node Maintenance Operator is supported in a restricted network environment.
...
-You can place a node into maintenance from the CLI or by using a `NodeMaintenance` CR.
+You can place a node into maintenance from the web console or in the CLI by using a `NodeMaintenance` CR.
 
-include::modules/eco-setting-node-maintenance-cli.adoc[leveloffset=+2]
+include::modules/eco-setting-node-maintenance-cr-web-console.adoc[leveloffset=+2]
 
-include::modules/eco-setting-node-to-maintenance-mode-with-cr.adoc[leveloffset=+2]
+include::modules/eco-setting-node-maintenance-cr-cli.adoc[leveloffset=+2]
 
 You can check the status of current `NodeMaintenance` CR tasks.
```
Comment (bobfuru) [inline nodes/nodes/eco-node-maintenance-operator.adoc:38]: Can you check this? I don't think you want it showing up at the end of the module. It shows up as the intro sentence in the module that follows. 
image

**Review by bobfuru — COMMENTED** [review]
It looks like you have a couple of redundant module files in the PR, with 2 module files that don't appear to be used anywhere in the PR. LMK if you need more clarifying details for my comments. Thanks!

File: modules/eco-setting-node-maintenance-cr-web-console.adoc [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+.Procedure
+
+. From the *Administrator* perspective in the web console, navigate to *Operators* → *Installed Operators*.
+
+. Select the Node Maintenance Operator from the list of operators.
+
+. In the *Node Maintenance* tab, click *Create NodeMaintenance*.
+
+. In the *Create NodeMaintenance* page, select to configure the `NodeMaintenance` CR via *Form view* or *YAML view*.
```
Comment (razo7) [inline modules/eco-setting-node-maintenance-cr-web-console.adoc:24]: Maybe add that the user needs to modify/update the CR for applying the CR on a real node :)

In the *Create NodeMaintenance* page, select to configure the `NodeMaintenance` CR for setting the node's name  via *Form view* or *YAML view*.

File: modules/eco-setting-node-maintenance-cr-cli.adoc (hunk 11 above)
Comment (razo7) [reply modules/eco-setting-node-maintenance-cr-cli.adoc:38]: I think it is already the same or I a missing something.
In step 1 we create a CR file, `nodemaintenance-cr.yaml`, where the CR's name is `maintenance-example` and in step 2 we apply/create the CR by it's file name, `nodemaintenance-cr.yaml`.

File: modules/eco-delete-nodemaintenance-cr-web-console.adoc (hunk 3 above)
Comment (razo7) [reply modules/eco-delete-nodemaintenance-cr-web-console.adoc:7]: I am confused as well

**Review by bobfuru — COMMENTED** [review]
Thanks for making the updates, @abhatt-rh! LGTM!

File: modules/eco-setting-node-maintenance-cr-cli.adoc (hunk 11 above)
Comment (abhatt-rh) [reply modules/eco-setting-node-maintenance-cr-cli.adoc:38]: Updated the metadata.name for consistency.

**Comment by bobfuru** [comment]
Looks great, @abhatt-rh - thanks for updating!

**Pull request merged**